lib/keycloak.rb

```diff
@@ -210,14 +210,16 @@ def self.get_token_introspection(token = '', client_id = '', secret = '', intros
       exec_request _request
     end
 
-    def self.url_login_redirect(redirect_uri, response_type = 'code', client_id = '', authorization_endpoint = '')
+    def self.url_login_redirect(redirect_uri, response_type = 'code', client_id = '', authorization_endpoint = '', scope: nil)
       verify_setup
 
       client_id = @client_id if isempty?(client_id)
       authorization_endpoint = @configuration['authorization_endpoint'] if isempty?(authorization_endpoint)
 
-      p = URI.encode_www_form(response_type: response_type, client_id: client_id, redirect_uri: redirect_uri)
-      "#{authorization_endpoint}?#{p}"
+      params = { response_type: response_type, client_id: client_id, redirect_uri: redirect_uri }
```

Comment:
```suggestion
      params = { response_type: response_type, client_id: client_id, redirect_uri: redirect_uri, scope: scope }
```
this should work as well?

Reply:
I think no, this shoud add scope to the URL if the scope value is nil and we don't want to set scope without value. Thus we don't set the scope the value is nil.

![Screenshot 2024-05-28 at 2 08 19 PM](https://github.com/Garaio-REM/keycloak/assets/57310683/e2ba6f7b-32af-4be3-9d92-4fd35f1165b9)